packages/js/components/src/table/index.tsx

```diff
@@ -0,0 +1,250 @@
+		}
+		if ( onQueryChange ) {
+			onQueryChange( 'paged' )( newPage, direction );
+		}
+	};
+
+	const allHeaders = headers;
+	const visibleHeaders = headers.filter( ( { key } ) =>
+		showCols.includes( key )
+	);
+	const visibleRows = rows.map( ( row ) => {
+		return headers
+			.map( ( { key }, i ) => {
+				return showCols.includes( key ) && row[ i ];
+			} )
+			.filter( Boolean );
+	} );
+	const classes = classnames( 'woocommerce-table', className, {
+		'has-actions': !! actions,
+		'has-menu': showMenu,
+		'has-search': hasSearch,
+	} );
+
+	return (
+		<Card className={ classes }>
+			<CardHeader>
+				<Text size={ 16 } weight={ 600 } as="h2" color="#23282d">
+					{ title }
+				</Text>
+				<div className="woocommerce-table__actions">{ actions }</div>
+				{ showMenu && (
+					<EllipsisMenu
+						label={ __(
+							'Choose which values to display',
+							'woocommerce'
+						) }
+						renderContent={ () => (
+							<Fragment>
+								{ /* @ts-expect-error: Ignoring the error until we migrate ellipsis-menu to TS*/ }
+								<MenuTitle>
+									{ /* @ts-expect-error: Allow string */ }
+									{ __( 'Columns:', 'woocommerce' ) }
+								</MenuTitle>
+								{ allHeaders.map(
+									( { key, label, required } ) => {
+										if ( required ) {
+											return null;
+										}
+										return (
+											<MenuItem
+												checked={ showCols.includes(
+													key
+												) }
+												isCheckbox
+												isClickable
+												key={ key }
+												onInvoke={
+													key !== undefined
+														? onColumnToggle( key )
+														: undefined
+												}
+											>
+												{ label }
+											</MenuItem>
+										);
+									}
+								) }
+							</Fragment>
+						) }
+					/>
+				) }
+			</CardHeader>
+			{ /* Ignoring the error to make it backward compatible for now. */ }
+			{ /* @ts-expect-error: size must be one of small, medium, largel, xSmall, extraSmall. */ }
+			<CardBody size={ null }>
+				{ isLoading ? (
+					<Fragment>
+						<span className="screen-reader-text">
+							{ __(
+								'Your requested data is loading',
+								'woocommerce'
+							) }
+						</span>
+						<TablePlaceholder
+							numberOfRows={ rowsPerPage }
+							headers={ visibleHeaders }
+							rowHeader={ rowHeader }
+							caption={ title }
+							query={ query }
+						/>
+					</Fragment>
+				) : (
+					<Table
+						rows={ visibleRows as TableCardProps[ 'rows' ] }
+						headers={
+							visibleHeaders as TableCardProps[ 'headers' ]
+						}
+						rowHeader={ rowHeader }
+						caption={ title }
+						query={ query }
+						onSort={
+							onSort ||
+							( onQueryChange( 'sort' ) as (
+								key: string,
+								direction: string
+							) => void )
+						}
+						rowKey={ rowKey }
+						emptyMessage={ emptyMessage }
+					/>
+				) }
+			</CardBody>
+
+			{ /* @ts-expect-error: justify is missing from the latest @types/wordpress__components */ }
```

Comment:
I'll try to see if I can make a PR to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/wordpress__components). The latest @types/wordpress__components doesn't have `justify` property typed.